### PR TITLE
types: reset the `MyDecimal` result buffer before putting arithmetic results into it

### PR DESCRIPTION
**Pull request by qw4990**
### What problem does this PR solve? 
Reset the `MyDecimal` result buffer before putting arithmetic results into it, which let it safer.

### What is changed and how it works?
Reset the `MyDecimal.wordBuf` before doing arithmetic operations.

### Check List 

Tests 

 - Unit test

### CODE REVIEW

File: util/chunk/column_test.go [hunk 1]
```diff
@@ -302,7 +302,8 @@ func (s *testChunkSuite) TestMyDecimal(c *check.C) {
 		}
 		c.Assert(d.Compare(&ds[i]), check.Equals, 0)
 
-		if err := types.DecimalAdd(&ds[i], d, &ds[i]); err != nil {
```
Comment (qw4990) [inline util/chunk/column_test.go]: See https://github.com/pingcap/tidb/blob/master/types/mydecimal.go#L1500

Comment (XuHuaiyu) [reply util/chunk/column_test.go]: Can we check `from1 != to && from2 != to` in DecimalAdd? It will be more safe.

Comment (qw4990) [reply util/chunk/column_test.go]: OK, I will add a check there.

File: types/mydecimal.go [hunk 2]
```diff
@@ -1518,7 +1518,17 @@ func DecimalSub(from1, from2, to *MyDecimal) error {
 	return doAdd(from1, from2, to)
 }
 
+func resetDecimal(d *MyDecimal) {
+	if d == nil {
+		return
+	}
+	for i := range d.wordBuf {
+		d.wordBuf[i] = 0
```
Comment (XuHuaiyu) [inline types/mydecimal.go]: Should the other member attributes of MyDecimal be reset?
We can not be sure whether we'll add any other functions that need to call resetDecimal.

File: types/mydecimal.go [hunk 3]
```diff
@@ -2000,6 +2012,7 @@ func DecimalMod(from1, from2, to *MyDecimal) error {
 }
 
 func doDivMod(from1, from2, to, mod *MyDecimal, fracIncr int) error {
+	resetDecimal(mod)
```
Comment (SunRunAway) [inline types/mydecimal.go]: should we reset `to` here?

File: types/mydecimal.go (hunk 2 above)
Comment (qw4990) [reply types/mydecimal.go]: What're your opinions about how to reset other attributes in `MyDecimal`?
How about setting them to zero values?

Comment (XuHuaiyu) [reply types/mydecimal.go]: That's what I mean.

File: util/chunk/column_test.go (hunk 1 above)
Comment (qw4990) [reply util/chunk/column_test.go]: Update, PTAL

File: types/mydecimal.go (hunk 3 above)
Comment (qw4990) [reply types/mydecimal.go]: My mistake, this check now has moved to `DecimalMod`.

File: types/mydecimal.go (hunk 2 above)
Comment (qw4990) [reply types/mydecimal.go]: fixed, PTAL.

**Pull request merged**